src/proofTypes/groth16/processor/index.ts

```diff
@@ -1,40 +1,69 @@
 import {
+  Formatter,
   Groth16VerificationKey,
   Groth16VerificationKeyInput,
   Proof,
   ProofInput,
 } from '../types';
-import * as formatter from '../formatter';
 import { ProofProcessor } from '../../../types';
+import { ProofOptions } from '../../../session/types';
 
 class Groth16Processor implements ProofProcessor {
   /**
-   * Formats the zk-SNARK proof based on the curve.
+   * Dynamically selects the appropriate formatter module based on the provided library option.
+   *
+   * @param {ProofOptions} options - The proof options containing the library type.
+   * @throws {Error} If the library is unsupported or the module cannot be loaded.
+   * @returns {Object} The formatter module corresponding to the specified library.
+   */
+  private getFormatter(options: ProofOptions): Formatter {
+    try {
+      // eslint-disable-next-line @typescript-eslint/no-require-imports
+      const formatter = require(`../formatter/${options.library}`) as Formatter;
```

Comment:
Mapping the library to the file path would be safer and more clear, in case there is unintentional file renaming.

Reply:
I just wanted to avoid more places to update things when additional libraries are added, just update the enum - it will already error if it cant find it and tell you why.  

I did go back and forth a bit on this though, just dont like having to update things all over the place and derive it from config enum name instead.